libraries/AP_Devo_Telem/AP_Devo_Telem.cpp

```diff
@@ -0,0 +1,119 @@
+// -*- tab-width: 4; Mode: C++; c-basic-offset: 4; indent-tabs-mode: nil -*-
+/*
+
+   Inspired by work done here https://github.com/PX4/Firmware/tree/master/src/drivers/DEVO_telemetry from Stefan Rado <px4@sradonia.net>
+
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+/* 
+   DEVO Telemetry library
+*/
+#include "AP_Devo_Telem.h"
+extern const AP_HAL::HAL& hal;
+
+DevoMPacket AP_DEVO_Telem::devoPacket = {DEVOM_SYNC_BYTE, 0};
+
+//constructor
+AP_DEVO_Telem::AP_DEVO_Telem(AP_AHRS &ahrs, AP_BattMonitor &battery) :
+    _ahrs(ahrs),
+    _battery(battery),
+    _port(NULL),
+    _initialised_uart(false),
+    _last_frame_ms(0)
+    {}
+
+// init - perform require initialisation including detecting which protocol to use
+void AP_DEVO_Telem::init(const AP_SerialManager& serial_manager)
+{
+    
+
+    // check for DEVO_DPort
+    if ((_port = serial_manager.find_serial(AP_SerialManager::SerialProtocol_Devo_Telem, 0))) {
+        _port->set_flow_control(AP_HAL::UARTDriver::FLOW_CONTROL_DISABLE);
+        // initialise uart
+        _port->begin(AP_SERIALMANAGER_DEVO_TELEM_BAUD, AP_SERIALMANAGER_DEVO_BUFSIZE_RX, AP_SERIALMANAGER_DEVO_BUFSIZE_TX);
+        _initialised_uart = true;   // SerialManager initialises uart for us
+    }
+}
+
+
+long AP_DEVO_Telem::gpsDdToDmsFormat(float ddm){
```

Comment:
"long" and "int" needs to be changed to an unambiguous standard type such as int32_t.